File: openbis_api/source/java/ch/ethz/sis/openbis/generic/dssapi/v3/dto/dataset/create/FullDataSetCreation.java
package ch.ethz.sis.openbis.generic.dssapi.v3.dto.dataset.create;

import java.util.List;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.create.DataSetCreation;
import ch.ethz.sis.openbis.generic.dssapi.v3.dto.datasetfile.create.DataSetFileCreation;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author anttil
 */
@JsonObject("dss.dto.dataset.create.FullDataSetCreation")
public class FullDataSetCreation implements ICreation
{
    private static final long serialVersionUID = 1L;

    private DataSetCreation metadataCreation;

    private List<DataSetFileCreation> fileMetadata;

    public DataSetCreation getMetadataCreation()
    {
        return metadataCreation;
    }

    public void setMetadataCreation(DataSetCreation metadataCreation)
    {
        this.metadataCreation = metadataCreation;
    }

    public List<DataSetFileCreation> getFileMetadata()
    {
        return fileMetadata;
    }

    public void setFileMetadata(List<DataSetFileCreation> fileMetadata)
    {
        this.fileMetadata = fileMetadata;
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("metadataCreation", metadataCreation).append("fileMetadata", fileMetadata).toString();
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dss/dto/dataset/create/FullDataSetCreation.js
define([ "stjs" ], function(stjs) {
	var FullDataSetCreation = function() {
	};
	stjs.extend(FullDataSetCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'dss.dto.dataset.create.FullDataSetCreation';
		constructor.serialVersionUID = 1;
		prototype.metadataCreation = null;
		prototype.fileMetadata = null;
		
		prototype.getMetadataCreation = function() {
			return this.metadataCreation;
		};
		prototype.setMetadataCreation = function(metadataCreation) {
			this.metadataCreation = metadataCreation;
		};
		prototype.getFileMetadata = function() {
			return this.fileMetadata;
		};
		prototype.setFileMetadata = function(fileMetadata) {
			this.fileMetadata = fileMetadata;
		};
	}, {
		metadataCreation : "DataSetCreation",
		fileMetadata : {
			name : "List",
			arguments : [ "Object" ]
		}
	});
	return FullDataSetCreation;
})